Gems/ROS2/Code/Source/RobotImporter/URDF/JointsMaker.cpp

```diff
@@ -89,35 +96,44 @@ namespace ROS2
                 followColliderEntity->Deactivate();
             }
             break;
-        case urdf::Joint::REVOLUTE:
+        case sdf::JointType::REVOLUTE:
             { // Hinge
                 jointComponent = followColliderEntity->CreateComponent<PhysX::EditorHingeJointComponent>();
                 followColliderEntity->Activate();
 
-                const double limitUpper = AZ::RadToDeg(joint->limits->upper);
-                const double limitLower = AZ::RadToDeg(joint->limits->lower);
-                AZ_Printf(
-                    "JointsMaker",
-                    "Setting limits : upper: %.1f lower: %.1f (URDF:%f,%f)",
-                    limitUpper,
-                    limitLower,
-                    joint->limits->upper,
-                    joint->limits->lower);
-                PhysX::EditorJointRequestBus::Event(
-                    AZ::EntityComponentIdPair(followColliderEntityId, jointComponent->GetId()),
-                    [&rotation, &limitLower, &limitUpper](PhysX::EditorJointRequests* editorJointRequest)
-                    {
-                        editorJointRequest->SetVector3Value(PhysX::JointsComponentModeCommon::ParameterNames::Rotation, rotation);
-                        editorJointRequest->SetLinearValuePair(
-                            PhysX::JointsComponentModeCommon::ParameterNames::TwistLimits,
-                            PhysX::AngleLimitsFloatPair(limitUpper, limitLower));
-                    });
+                if (jointAxis != nullptr)
+                {
+                    using LimitType = decltype(jointAxis->Upper());
+                    const double limitUpper = jointAxis->Upper() != AZStd::numeric_limits<LimitType>::infinity()
+                        ? AZ::RadToDeg(jointAxis->Upper())
+                        : AZ::RadToDeg(AZ::Constants::TwoPi);
+                    const double limitLower = jointAxis->Lower() != -AZStd::numeric_limits<LimitType>::infinity()
+                        ? AZ::RadToDeg(jointAxis->Lower())
+                        : -AZ::RadToDeg(AZ::Constants::TwoPi);
```

Comment:
Suggestion: Consider adding a comment here explaining that in PhysX the Revolute joint is limited to a max of +/- 2 pi for the joint limits. Also might want to consider clamping the jointAxis values as well to ensure they stay within the range, and/or printing a warning if we got input values outside of the allowed range.

Reply:
Also... if both upper & lower are set to +/- infinity, should we create a CONTINUOUS joint instead? (I saw in a different file that you mentioned that continuous joints were getting converted to revolute joints with infinite limits)

Reply:
Well the parsed sdf::Root object is constant, so we can't change the type of the joint to CONTINUOUS, so we parse that joint is revolute and then perform the same logic as the continuous block right above.

Reply:
Oh, I see, the logic is _almost_ identical between the two. This would need to set EnableLimits to false to be equivalent and turn it back into a continuous joint though.

Reply:
Wait it's _almost_ identical?
It should actually be identical when the limit values are (-∞, ∞)
So in that case the Limits need to be turned off.

That needs to be fixed.